File: 3033/main.go
package main

import "fmt"

func modifiedMatrix(matrix [][]int) [][]int {
	m := make([]int, len(matrix[0]))

	for i, mat  := range matrix {
		for j := range mat {
			if matrix[i][j] > m[j] {
				m[j] = matrix[i][j]
			}
		}
	}
	for i, mat  := range matrix {
		for j := range mat {
			if matrix[i][j] == -1 {
				matrix[i][j] = m[j]
			}
		}
	}
	return matrix
}

func main() {
	fmt.Println(modifiedMatrix([][]int{
		{1, 2, -1},
		{4, -1, 6},
		{7, 8, 9},
	}))
}

File: 3033/main.rs
struct Solution {}

impl Solution {
    pub fn modified_matrix(mut matrix: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let n = matrix.len();
        let m = matrix[0].len();
        for j in 0..m {
            let mut zd = -1;
            for i in 0..n {
                zd = std::cmp::max(zd, matrix[i][j])
            }
            for i in 0..n {
                if matrix[i][j] == -1 {
                    matrix[i][j] = zd
                }
            }
        }
        matrix
    }
}

fn main() {
    println!("{:#?}", Solution::modified_matrix(vec![
        vec![1,2,-1],
        vec![4,-1, 6],
        vec![7,8,9]
    ]))
}